chainer/functions/activation/softmax.py

```diff
@@ -36,17 +37,17 @@ def forward(self, x):
             one = numpy.array(1, dtype=oz_dtype).ctypes
             zero = numpy.array(0, dtype=oz_dtype).ctypes
             handle = cudnn.get_handle()
-            x_cube = x[0].reshape(x[0].shape[:2] + (-1, 1))
+            x_cube = x[0].reshape(self._get_cube_shape(x[0].shape))
             desc = cudnn.create_tensor_descriptor(x_cube)
             self.y = xp.empty_like(x[0])
             libcudnn.softmaxForward(
                 handle, _algorithm, _mode, one.data, desc.value,
                 x_cube.data.ptr, zero.data, desc.value,
                 self.y.data.ptr)
         else:
```

Comment:
(Sorry I added this comment here because Github doesn't allow me to comment at L:34)
cuDNN only supports two softmax modes:

- CUDNN_SOFTMAX_MODE_INSTANCE:
    - The softmax operation is computed per image (N) across the dimensions C,H,W.
- CUDNN_SOFTMAX_MODE_CHANNEL:
    - The softmax operation is computed per spatial location (H,W) per image (N) across the dimension C.

The former one corresponds to the case when `axis=(1, 2, 3)` (if the Chainer's softmax can accept multiple axes), and the latter is the normal softmax. So in all the other cases, we can't use cuDNN to perform softmax.

Thus, please add a condition to the `if` statement at L:34 like `self.axis >2`.